src/types/server-types.ts

```diff
@@ -139,47 +56,23 @@ export interface ActionDefinition {
   dynamicBranchInput?: string;
   /**An example of the payload outputted by an Action*/
   examplePayload?:
-    | PerformDataStructureReturn
-    | PerformBranchingDataStructureReturn;
-}
-
-/** Action-specific Display attributes. */
-interface ActionDisplayDefinition extends DisplayDefinition {
-  /** Directions to help guide the user if additional configuration is required for this Action. */
-  directions?: string;
-  /** Indicate that this Action is important and/or commonly used from the parent Component. Should be enabled sparingly. */
-  important?: boolean;
-}
-
-type ActionLoggerFunction = (...args: unknown[]) => void;
-
-interface ActionLogger {
-  debug: ActionLoggerFunction;
-  info: ActionLoggerFunction;
-  log: ActionLoggerFunction;
-  warn: ActionLoggerFunction;
-  error: ActionLoggerFunction;
-}
-
-/** Context provided to perform method containing helpers and contextual data */
-interface ActionContext {
-  /** Credential for the action, optional since not all actions will require a credential */
-  credential?: Credential;
-  /** Logger for permanent logging; console calls are also captured */
-  logger: ActionLogger;
-  /** A key/value store that may be used to store small amounts of data that is persisted between Instance executions */
-  instanceState: Record<string, unknown>;
-  /** A unique id that corresponds to the step on the Integration */
-  stepId: string;
+    | ServerPerformDataStructureReturn
+    | ServerPerformBranchingDataStructureReturn;
 }
 
 /** Collection of input parameters provided by the user or previous steps' outputs */
 interface ActionInputParameters {
-  [key: string]: any;
+  [key: string]:
+    | boolean
```

Comment:
I think if you include `unknown` in a union like this it just compiles down to `unknown`. That means we could just change this to

```typescript
interface ActionInputParameters {
  [key: string]: unknown;
}
```

Reply:
Good call!